### PR TITLE
Add sample for client filtering

### PR DESCRIPTION
**Pull request by kul3r4**
This adds a paragraph explaining how to do filtering and sorting with amp-list and amp-bind.

### CODE REVIEW

**Comment by kul3r4** [comment]
@sebastianbenz this can be reviewed

File: src/10_Introduction/AMP_for_E-Commerce_Getting_Started.html [hunk 1]
```diff
@@ -234,5 +234,30 @@
...
+    <amp-state id="allProducts" src="/json/related_products.json"></amp-state>
+    <h3>Select your favourite color</h3>
+    <select class="mt1 ampstart-input" on="change:AMP.setState({filteredProducts: allProducts.items.filter(a => event.value == 'all' ? true : a.color == event.value)})">
+        <option value="all" selected="">All</option>
+        <option value="red">red</option>
+        <option value="green">green</option>
+        <option value="yellow">yellow</option>
+        <option value="orange">orange</option>
+    </select>
+    <amp-list credentials="include" width="auto" height="100" layout="fixed-height" src="/json/related_products.json" [src]="filteredProducts" >
```
Comment (sebastianbenz) [inline src/10_Introduction/AMP_for_E-Commerce_Getting_Started.html]: Can you make the initial height match the number of items + update the height based on the filter results.

Comment (kul3r4) [reply src/10_Introduction/AMP_for_E-Commerce_Getting_Started.html]: done

**Comment by kul3r4** [comment]
@sebastianbenz I updated the sample.
Where this should live according to you? Advanced or Dynamic AMP section?

**Comment by sebastianbenz** [comment]
I'd say Dynamic AMP.

**Comment by kul3r4** [comment]
@sebastianbenz PTAL

**Comment by kul3r4** [comment]
@sebastianbenz ping

**Comment by sebastianbenz** [comment]
Can you fix the merge conflicts?

**Comment by sebastianbenz** [comment]
(and sorry for forgetting to review)

**Comment by kul3r4** [comment]
Done!

File: src/40_Dynamic_AMP/Products_Filter.html [hunk 2]
```diff
@@ -0,0 +1,65 @@
+<!-- ## Products Filter -->
+<!--
+This is a sample showing how to implement products filter.
```
Comment (sebastianbenz) [inline src/40_Dynamic_AMP/Products_Filter.html]: Same as above.

File: src/40_Dynamic_AMP/Products_Filter.html [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+  <style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style><noscript><style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript>
+
+</head>
+<body>
+  <!--
+    ## Client-side Filter
+
+    It is possible to implement client-side filtering by using `amp-list`, `amp-state` and `amp-bind`. The `amp-state` is initially
+    setup with data from an endpoint then everytime the user select a choice the variable `filteredProducts` is set
```
Comment (sebastianbenz) [inline src/40_Dynamic_AMP/Products_Filter.html]: This whole sentence is a bit complicated: "The `amp-state` is initially setup..."

File: src/40_Dynamic_AMP/Products_Filter.html [hunk 4]
```diff
@@ -0,0 +1,65 @@
+<!-- ## Products Filter -->
```
[the remark this answers is not in the document]
Comment (kul3r4) [reply src/40_Dynamic_AMP/Products_Filter.html]: I think Client-side filtering is enough, we don't need the "with...." as we explained it later

File: src/40_Dynamic_AMP/Products_Filter.html (hunk 2 above)
Comment (kul3r4) [reply src/40_Dynamic_AMP/Products_Filter.html]: same as above

File: src/40_Dynamic_AMP/Products_Filter.html (hunk 3 above)
Comment (kul3r4) [reply src/40_Dynamic_AMP/Products_Filter.html]: Updated.

File: src/40_Dynamic_AMP/Client-side filtering.html [hunk 5]
```diff
@@ -0,0 +1,70 @@
+<!-- ## Products Filter -->
```
Comment (sebastianbenz) [inline src/40_Dynamic_AMP/Client-side filtering.html]: Can you rename this heading to introduction

File: src/40_Dynamic_AMP/Client-side filtering.html [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+</head>
+<body>
+  <!--
+    ## Client-side Filter
+
+    It is possible to implement client-side filtering by using `amp-list`, `amp-state` and `amp-bind`. 
+    
+    The `amp-state` is initially setup with data from an endpoint which returns a list of available products; 
+    the user can choice between different colors and that selection sets the variable `filteredProducts` to the result of a filter expression.
+    The filter expression is a javascript expression which uses the `Array.filter` function.
```
Comment (sebastianbenz) [inline src/40_Dynamic_AMP/Client-side filtering.html]: nit: it's not a JS expression, it's an amp-bind expression which looks like JS

**Comment by kul3r4** [comment]
Thanks, updated those!

File: src/40_Dynamic_AMP/Client-side filtering.html [hunk 7]
```diff
@@ -1,4 +1,4 @@
-<!-- ## Products Filter -->
+<!-- ## Client-side Filter -->
```
Comment (sebastianbenz) [inline src/40_Dynamic_AMP/Client-side filtering.html]: We usually use 'Introduction' here

**Comment by kul3r4** [comment]
Right, update it

**Pull request merged**